api/cds.proto

```diff
@@ -0,0 +1,233 @@
+syntax = "proto3";
+
+import "api/address.proto";
+import "api/node.proto";
+import "api/ssl_context.proto";
+
+import "google/protobuf/wrappers.proto";
+
+service ClusterDiscoveryService {
+  // Return list of all clusters, this proxy will load balance to.
+  rpc StreamClusters(ClusterDiscoveryRequest)
+      returns (stream ClusterDiscoveryResponse) {
+  }
+}
+
+message ClusterDiscoveryRequest {
+  Node node = 1;
+}
+
+message ClusterDiscoveryResponse {
+  repeated Cluster cluster = 1;
+}
+
+message Duration {
+  google.protobuf.UInt32Value milliseconds = 1;
+}
+
+message HealthCheck {
+  // The time to wait for a health check response. If the timeout is reached the
+  // health check attempt will be considered a failure.
+  Duration timeout = 1;
+  // The interval between health checks.
+  Duration interval = 2;
+  // An optional jitter amount in millseconds. If specified, during every
+  // internal Envoy will add 0 to interval_jitter to the wait time.
+  Duration interval_jitter = 3;
+
+  // The number of unhealthy health checks required before a host is marked
+  // unhealthy. Note that for http health checking if a host responds with 503
+  // this threshold is ignored and the host is considered unhealthy immediately.
+  google.protobuf.UInt32Value unhealthy_threshold = 4;
+  // The number of healthy health checks required before a host is marked
+  // healthy. Note that during startup, only a single successful health check is
+  // required to mark a host healthy.
+  google.protobuf.UInt32Value healthy_threshold = 5;
+
+  // Non-serving port for health checking.
+  google.protobuf.UInt32Value alt_port = 6;
+  // Reuse health check connection between health checks. Default is true.
+  google.protobuf.BoolValue reuse_connection = 7;
+
+  // Describes the encoding of the payload bytes in the payload
+  message Payload {
+    oneof payload {
+      google.protobuf.StringValue text = 1;
+      google.protobuf.BytesValue binary = 2;
+    }
+  }
+  message HttpHealthCheck {
+    // TODO(htuch): Figure out what this is.
```

Comment:
this is the same as GCE HealthCheck HTTP host - https://cloud.google.com/compute/docs/reference/latest/healthChecks

The value of the host header in the HTTPS health check request. If left empty (default value), the IP on behalf of which this health check is performed will be used.